operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -0,0 +1,309 @@
+        allEntries.add(patientEntry);
+        
+        // We can't always use the "date" query parameter to query by clinical date, only with some resources.
+        // Initial list obtained from the github issue: https://github.com/IBM/FHIR/issues/1044#issuecomment-769788097
+        // Otherwise the search throws an exception. We create a params map with and without and use as needed
+        MultivaluedMap<String, String> queryParameters = parseQueryParameters(parameters);
+        MultivaluedMap<String, String> queryParametersWithoutDates = new MultivaluedHashMap<String, String>(queryParameters);
+        queryParametersWithoutDates.remove(DATE_QUERY_PARAMETER);
+        
+        List<String> resourceTypesOverride = getOverridenIncludedResourceTypes(parameters);
+        List<String> resourceTypes = resourceTypesOverride.isEmpty() ? defaultResourceTypes : resourceTypesOverride;
+        
+        for (String compartmentType : resourceTypes) {
+            MultivaluedMap<String, String> searchParameters = queryParameters;
+            if (!SUPPORT_CLINICAL_DATE_QUERY.contains(compartmentType)) {
+                searchParameters = queryParametersWithoutDates;
+            }
+            Bundle results = null;
+            try {
+                results = resourceHelper.doSearch(compartmentType, PATIENT, logicalId, searchParameters, null, null, null);
+            } catch (Exception e) {
+                LOG.warning("Error retrieving $everything resources of type " + compartmentType + " for patient " + logicalId);
+                continue;
+            }
+            allEntries.addAll(results.getEntry());
+        }
+        
+        Bundle.Builder bundleBuilder = Bundle.builder()
+                .type(BundleType.SEARCHSET)
+                .id(UUID.randomUUID().toString())
+                .entry(allEntries)
+                .total(UnsignedInt.of(allEntries.size()));        
+        
+        Parameters outputParameters;
+        try {
+            outputParameters = FHIROperationUtil.getOutputParameters(bundleBuilder.build());
+        } catch (Exception e) {
+            FHIROperationException exceptionWithIssue = buildExceptionWithIssue("An unexpected error occurred while creating the operation output parameters for the resulting Bundle.", IssueType.EXCEPTION);
+            LOG.throwing(this.getClass().getName(), "doInvoke", exceptionWithIssue);
+            throw exceptionWithIssue;
+        }
+        LOG.exiting(this.getClass().getName(), "doInvoke", outputParameters);
+        return outputParameters;
+    }
+
+    /**
+     * Parse the parameters and turn them into a {@link MultivaluedMap} to pass to the search service
+     * 
+     * @param parameters the operation parameters
+     * @return the {@link MultivaluedMap} for the search service built from the parameters 
+     */
+    protected MultivaluedMap<String, String> parseQueryParameters(Parameters parameters) {
```

Comment:
May want to look at (or call) the validation done in AbstractOperation.validateInputParameters() - provides good detail on errors. Based on what mode you're in (strict vs. lenient - see my comment in getOverridenIncludeResourceTypes()), either just log the error or throw an exception.

Reply:
I am not doing any validation here, if I understand correctly the validation already happened by the time we get here. I am just converting `$everything` query parameters to `$search` query parameters here, maybe I am missing something?

Reply:
My bad - you're correct.